### ISSUE TITLE
Safer event callbacks

### ISSUE THREAD

**Issue by maicki** [maicki/react-native-event-bridge#9]
Currently we pass an array as callback from native to react native. This makes it not really clear in what order parameters should be passed in. We assume that the first parameter is an error and the second is the data, but it's not clear from the API that this is the case. We should define specifically what should be passed in into the callback and in what order.

Furthermore as the callback can be called with nil arguments we have to change the expected parameters on the JavaScript side to be optional.

Some work is going at: https://github.com/maicki/react-native-event-bridge/tree/MSSaferCallback

@thorbenprimke Based on the changes in the `MSSaferCallback` branch on iOS you have a good idea how we could handle this on the Android side?

**Comment by thorbenprimke** [comment]
On the iOS / Android side, what about having a calls like 

```
MSREventBridge
```

that would have two methods: 

```
onSuccess(Object data)
onFailure(Object error)
```

Unless there is a case where you think one would have to pass both an error and data. 

This class would simply wrap the `Callback`. 

👍 to the new type on the JS side.

**Comment by maicki** [comment]
Yep thought about something like that, sounds great! On the library side we would handle the safety and pass down null for either the data or error depending on what was called on the callback!

**Comment by thorbenprimke** [comment]
Hmm. I'm not sure I'm a fan of that (two callbacks) - because again you just call invoke. Having more explicitly named methods I think just makes it easier to use (and it would correspond better to the typed JS function).

**Comment by maicki** [comment]
Resolved via #11
